src/minimap.cpp

```diff
@@ -0,0 +1,476 @@
+		if (it != m_blocks_cache.end()) {
+			MinimapPixel *pixel = &it->second->data[relpos.X + relpos.Z * MAP_BLOCKSIZE];
+			if (pixel->id != CONTENT_AIR) {
+				pixel_height += pixel->height;
+				return pixel;
+			}
+		}
+		pixel_height -= MAP_BLOCKSIZE;
+	}
+	return NULL;
+}
+
+s16 MinimapUpdateThread::getAirCount (v3s16 pos, s16 height)
+{
+	s16 air_count = 0;
+	v3s16 blockpos_max, blockpos_min, relpos;
+	getNodeBlockPosWithOffset(v3s16(pos.X, pos.Y - height / 2, pos.Z), blockpos_min, relpos);
+	getNodeBlockPosWithOffset(v3s16(pos.X, pos.Y + height / 2, pos.Z), blockpos_max, relpos);
+	std::map<v3s16, MinimapMapblock *>::iterator it;
+	for (s16 i = blockpos_max.Y; i > blockpos_min.Y - 1; i--) {
+		it = m_blocks_cache.find(v3s16(blockpos_max.X, i, blockpos_max.Z));
+		if (it != m_blocks_cache.end()) {
+			MinimapPixel *pixel = &it->second->data[relpos.X + relpos.Z * MAP_BLOCKSIZE];
+			air_count += pixel->air_count;
+		}
+	}
+	return air_count;
+}
+
+void MinimapUpdateThread::getMap (v3s16 pos, s16 size, s16 height, bool radar)
+{
+	v3s16 p = v3s16 (pos.X - size / 2, pos.Y, pos.Z - size / 2);
+
+	for(s16 x = 0; x < size; x++) {
+		for(s16 z = 0; z < size; z++){
+			u16 id = CONTENT_AIR;
+			MinimapPixel* minimap_pixel = &data->minimap_scan[x + z * size];
+			if (!radar) {
+				s16 pixel_height = 0;
+				MinimapPixel* cached_pixel = 
+					getMinimapPixel(v3s16(p.X + x, p.Y, p.Z + z), height, pixel_height);
+				if (cached_pixel) {
+					id = cached_pixel->id;
+					minimap_pixel->height = pixel_height;
+				}
+			} else {
+				minimap_pixel->air_count = getAirCount (v3s16(p.X + x, p.Y, p.Z + z), height);
+			}
+			minimap_pixel->id = id;
+		}
+	}
+}
+
+Mapper::Mapper(IrrlichtDevice *device, Client *client):
+	m_minimap_update_thread(MinimapUpdateThread(device, client))
+{
+	this->device	= device;
+	this->client	= client;
+	this->driver	= device->getVideoDriver();
+	this->tsrc		= client->getTextureSource();
+	this->player	= client->getEnv().getLocalPlayer();
+	this->shdrsrc   = client->getShaderSource();
+
+	m_enable_shaders = g_settings->getBool("enable_shaders");
+	data = new MinimapData;
+	data->mode = MINIMAP_OFF;
+	data->radar = false;
+	data->map_invalidated = true;
+	data->heightmap_image = NULL;
+	data->minimap_image = NULL;
+	data->texture = NULL;
+	std::string fname1, fname2;
+	if (g_settings->getBool("minimap_shape_round")) {
+		fname1 = "minimap_mask_round.png";
+		fname2 = "minimap_overlay_round.png";
+		data->minimap_shape_round = true;
+	} else {
+		fname1 = "minimap_mask_square.png";
+		fname2 = "minimap_overlay_square.png";
+		data->minimap_shape_round = false;	
+	}
+	data->minimap_mask = driver->createImage (tsrc->getTexture(fname1),
+			core::position2d<s32>(0,0),	core::dimension2d<u32>(512,512));
+	data->minimap_overlay = tsrc->getTexture(fname2);
+	data->player_marker = tsrc->getTexture("player_marker.png");	
+
+	m_meshbuffer = getMinimapMeshBuffer();
+	m_minimap_update_thread = MinimapUpdateThread(device, client);
```

Comment:
/home/nerzhul/Devel/minetest/src/minimap.cpp:244:26: error: object of type 'MinimapUpdateThread' cannot be assigned because its copy assignment operator is implicitly deleted
        m_minimap_update_thread = MinimapUpdateThread(device, client);

Same error, and also you affect this variable twice